### PR TITLE
CORDA-499: Dokka 'Kt' class cleanup

### PR DESCRIPTION
**Pull request by rnicoll**
Continued cleanup of Kotlin structure to eliminate (where possible) generated `...Kt` classes.

### CODE REVIEW

File: client/mock/src/main/kotlin/net/corda/client/mock/Generator.kt [hunk 1]
```diff
@@ -58,16 +57,42 @@ class Generator<out A>(val generate: (SplittableRandom) -> Try<A>) {
...
+         */
         fun <A> choice(generators: List<Generator<A>>) = intRange(0, generators.size - 1).flatMap { generators[it] }
 
         fun <A> success(generate: (SplittableRandom) -> A) = Generator { Try.Success(generate(it)) }
+        /**
+         * Pick a generator from the specified list, with a probability assigned to each generator, then run the
+         * chosen generator.
+         *
+         * @param generators a list of probabilities of a generator being chosen, and generators. Probabilities must be
+         * non-negative, and should total to 1.0.
```
Comment (exFalso) [inline client/mock/src/main/kotlin/net/corda/client/mock/Generator.kt]: They need not total `1.0`, the values are "normalized" implicitly in the binary search. We shouldn't require this, as certain probabilities cannot be exactly represented with ieee doubles totalling 1.

File: client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt [hunk 2]
```diff
@@ -393,6 +406,19 @@ object RpcClientObservableSerializer : Serializer<Observable<*>>() {
         return serializationContext.withProperty(RpcObservableContextKey, observableContext)
     }
 
+    private fun <T> pinInSubscriptions(observable: Observable<T>, hardReferenceStore: MutableSet<Observable<*>>): Observable<T> {
```
Comment (exFalso) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt]: This should be in the companion.
Are we moving all global functions into classes?

File: core/src/main/kotlin/net/corda/core/messaging/FlowHandle.kt [hunk 3]
```diff
@@ -57,16 +57,11 @@ data class FlowProgressHandleImpl<A>(
...
+            progress.subscribe({}, {}).unsubscribe()
+        } catch (e: Exception) {
+            // Swallow any other exceptions as well.
+        }
         returnValue.cancel(false)
     }
-}
-
-// Private copy of the version in client:rpc.
-private fun <T> Observable<T>.notUsed() {
```
Comment (exFalso) [inline core/src/main/kotlin/net/corda/core/messaging/FlowHandle.kt]: Why was this removed? Isn't this something we encourage users to use?

File: client/mock/src/main/kotlin/net/corda/client/mock/Generator.kt (hunk 1 above)
Comment (rnicoll) [reply client/mock/src/main/kotlin/net/corda/client/mock/Generator.kt]: I did say "should", although accept that in a sentence after a "must" is a good way to be confusing. Might just remove that.

File: client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt (hunk 2 above)
Comment (rnicoll) [reply client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt]: I'm trying to remove the generated `Kt` classes, and finding a lot of cases where we have single-use global functions, in which case the simplest solution is to merge it into the class that uses it.

File: core/src/main/kotlin/net/corda/core/messaging/FlowHandle.kt (hunk 3 above)
Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/messaging/FlowHandle.kt]: We're keeping the underlying extension, but in this case it can only be used from `FlowHandle` (it's `private`), so again the simplest way to solve an empty `FlowHandleKt` class appearing in the Javadocs was to merge it into the calling code.

**Comment by rnicoll** [comment]
Removed note about probabilities summing to 1.0. Let me know if there's better solutions to avoiding creating `Kt` classes, but in most cases I'm not sure  these global functions are actually global.
